python/ray/serve/_private/http_proxy.py

```diff
@@ -670,8 +677,8 @@ async def send_request_to_replica_unary(
                 await asyncio.sleep(backoff_period)
         else:
             error_message = f"Task failed with {HTTP_REQUEST_MAX_RETRIES} retries."
-            await Response(error_message, status_code=500).send(scope, receive, send)
-            return "500"
+            await Response(error_message, status_code=408).send(scope, receive, send)
+            return "408"
```

Comment:
this doesn't look appropriate? this path is for the request failing after `MAX_RETRIES` retries

anyways I'd mostly worry about the new streaming path

Reply:
Yea, there are so much things going on in those code path got confused 😅